### PR TITLE
Spawngroup castables

### PR DESCRIPTION
**Pull request by norrismiv**
This PR allows for easily defined castables on a spawngroup with the following types:

`<Offense> <Defense> <NearDeath> <OnDeath>`

example xml added to a spawn:

```
      <Castables>
        <Offense>
          <Castable Name="srad" MinDmg="10" MaxDmg="20" Interval="10" Target="Attacker"/>
        </Offense>
      </Castables>
```

Need a second set of eyes on this though, as it appears to be working locally.

### CODE REVIEW

File: XML/XSD/Item.xsd [hunk 1]
```diff
@@ -183,7 +183,7 @@
   </xs:complexType>
 
   <xs:complexType name="Equipment">
-    <xs:attribute name="Slot" use="required" type="hyb:EquipmentSlot" default="None"/>
+    <xs:attribute name="Slot" use="required" type="hyb:EquipmentSlot"/>
```
Comment (Caeldeth) [inline XML/XSD/Item.xsd:186]: Why is the default being removed?

File: XML/XSD/Common.xsd [hunk 2]
```diff
@@ -432,6 +432,17 @@
     </xs:restriction>
   </xs:simpleType>
 
+  <xs:simpleType name="TargetType">
+    <xs:restriction base="xs:token">
+      <xs:enumeration value="None"/>
+      <xs:enumeration value="Self" />
+      <xs:enumeration value="Attacker" />
+      <xs:enumeration value="Group" />
+      <xs:enumeration value="Nearby" />
```
Comment (Caeldeth) [inline XML/XSD/Common.xsd:441]: Based on previous insanity in other xml, what is the definition of "Nearby"?  Like, I can see this making sense here, but is it numerically defined as, idk: everything in a 3x3 square centered on creature?

Comment (norrismiv) [reply XML/XSD/Common.xsd:440]: in this instance, nearby would be limited to the 1x1 square surrounding the creature. Note: currently unused and easily removed if needed.

File: XML/XSD/Item.xsd (hunk 1 above)
Comment (norrismiv) [reply XML/XSD/Item.xsd]: can't have `required` with a default.

**Comment by norrismiv** [comment]
@baughj this should now be ready for merge

**Pull request merged**